File: benchmark/corpus/03_div_zero/d_guarded_one_path.py
# The divisor is guarded on the strict path and not on the other; input takes the other.
import sys


def share(total, count, strict):
    if strict:
        if count == 0:
            return 0
        return total // count
    return total // count  # DANGER


line = sys.stdin.readline().strip()
try:
    count = int(line)
except ValueError:
    print("not a number: " + line)
else:
    print("each pays %d" % share(59500, count, False))

File: benchmark/corpus/03_div_zero/d_guarded_one_path.js
// The divisor is guarded on the strict path and not on the other; input takes the other.
function share(total, count, strict) {
  if (strict) {
    if (count === 0) return 0;
    return Math.floor(total / count);
  }
  return Math.floor(total / count); // DANGER
}

const line = (await new Response(Deno.stdin.readable).text()).trim();
const count = Number(line);
if (Number.isNaN(count)) console.log("not a number: " + line);
else console.log("each pays " + share(59500, count, false));
